net/libreswan/files/etc/config/libreswan

```diff
@@ -0,0 +1,37 @@
+config libreswan 'globals'
+	option debug '0' # set debug mode none/all
+	list virtual_private '10.0.0.0/8'
+	list virtual_private '192.168.0.0/16'
+	list virtual_private '172.16.0.0/12'
+	list virtual_private '25.0.0.0/8'
+	list virtual_private '100.64.0.0/10'
+	list virtual_private '!100.64.0.0/24'  # the address ranges that may live behind a NAT router through which a client connects
+	# option listen '192.168.2.100' # listening address, if set listen_interface would not be used
+	# option listen_interface 'wan' # listening interface
+	# option uniqueids 'yes' # yes/no
+
+config crypto_proposal 'p1'
+	option encryption_algorithm '3des' # possible values: 3des, aes, aes_ctr, aes_cbc, aes128, aes192, aes256, camellia_cbc
+	option hash_algorithm 'md5' # possible values: md5, sha1, sha256, sha384, sha512
+	list dh_group 'modp1536' # possible values: modp1536, modp2048, modp3072, modp4096, modp6144, modp8192, dh19, dh20, dh21, dh22, dh31
+
+# config tunnel 'vti2_1_5'
+# 	option left '%wan'  # left can be interface of ipaddress
+# 	option leftid '@left' # local id
+# 	option right '192.168.2.201' # remote endpoint public ip
+# 	option rightid '@62dd3e3f82339b002405245b' # rightid
+# 	option auto 'start' # what  operation, should be done automatically at IPsec startup
+# 	option authby 'secret' # how  the  two security gateways should authenticate each other
+# 	option psk 'AyG9RlTtQJIUxgxG' # preshare key
+# 	option ikev2 '1' # ike version
```

Comment:
so it means that if here is 0, ikev1 is the default  one ?

Reply:
yes

Reply:
Wouldn't it be better to use an option here `option ike` rather than a boolean? It may be that there will be an ikev3 at some point.

Reply:
This might still need rework in the future with ikev3 because the option `ikev2` is kept in sync with the underlying `libreswan` daemon ipsec configuration options.

Libreswan `ipsec.conf`, expects `ikev2=yes` or `ikev2=no`, so in the future, if ikev3 is introduced, it would still depend on new config options.